pkg/cli/clusteroperator/pernodepod/runtime.go

```diff
@@ -143,7 +143,24 @@ func (r *PerNodePodRuntime) Run(ctx context.Context, prePodHookFn PrePodHookFunc
 	return utilerrors.NewAggregate(errs)
 }
 
-func (r *PerNodePodRuntime) RestartNode(ctx context.Context, createPodFn CreatePodFunc, namespaceName string, node *corev1.Node) error {
+func (r *PerNodePodRuntime) HandleNode(ctx context.Context, createPodFn CreatePodFunc, namespaceName string, node *corev1.Node) error {
+	notReadyMessage := ""
+	for _, condition := range node.Status.Conditions {
+		if condition.Type != corev1.NodeReady {
+			continue
+		}
+		if condition.Status == corev1.ConditionTrue {
+			break
+		}
+		notReadyMessage = fmt.Sprintf("%s since %v - %s: %s", condition.Status, condition.LastTransitionTime.Format(time.RFC3339), condition.Reason, condition.Message)
+		break
+	}
+	if len(notReadyMessage) > 0 {
+		retErr := fmt.Errorf("nodes/%v is not ready, %v", node.Name, notReadyMessage)
+		fmt.Fprintln(r.ErrOut, retErr.Error())
```

Comment:
Is there a reason to both print an return the error here?  That usually results in double logging unnecessarily.

Reply:
> Is there a reason to both print an return the error here? That usually results in double logging unnecessarily.

It does, it's not perfect.  However if you don't report until the command finishes, there is a long feedback delay.  Given enough nodes, it's probably worthwhile to collect just the errors at the end anyway.  With only 6, it does look weird.